src/main/java/seedu/foodrem/logic/parser/itemcommandparser/ViewCommandParser.java

```diff
@@ -22,25 +20,7 @@ public class ViewCommandParser implements Parser<ViewCommand> {
      */
     public ViewCommand parse(String args) throws ParseException {
         requireNonNull(args);
-        Index index;
-
-        String trimmedArgument = args.trim();
-
-        if (!StringUtil.isInteger(trimmedArgument)) {
-            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.getUsage()));
-        }
-
-        if (!StringUtil.isNonZeroUnsignedInteger(trimmedArgument)) {
-            throw new ParseException("The index should be a non-negative number.");
-        }
-
-        try {
-            index = ParserUtil.parseIndex(args);
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.getUsage()),
-                    pe);
-        }
-
+        Index index = StringUtil.validateAndGetIndexFromString(args, ViewCommand.getUsage());
```

Comment:
Should this be `args.trim()` instead?

```suggestion
        Index index = StringUtil.validateAndGetIndexFromString(args.trim(), ViewCommand.getUsage());
```

Reply:
The trimming is already done inside but yes, this would be more defensive